soc/xtensa/intel_adsp/common/soc_mp.c

```diff
@@ -218,17 +193,17 @@ void z_mp_entry(void)
 	 * isn't using yet.  Manual inspection of generated code says
 	 * we're safe, but really we need a better solution here.
 	 */
-#ifndef CONFIG_SOC_SERIES_INTEL_CAVS_V25
-	z_xtensa_cache_flush_inv_all();
-#endif
+	if (!IS_ENABLED(CONFIG_SOC_SERIES_INTEL_CAVS_V25)) {
```

Comment:
is there any rule / recommendation when to use preprocessor and when (pseudo) run-time?

Reply:
Pretty much kernel style: use IS_ENABLED() everywhere you can because it's more readable.  Use #if's only where needed because the included code won't always be able to build successfully without the needed configuration (and even then we try to make sure that doesn't happen by e.g. not having preprocessor-dependent symbols defined, etc....)